### DISCUSSION TITLE
Purpose of recommending hash routing

### DISCUSSION THREAD

**Discussion by thejoyfuldev** [wailsapp/wails#3065] [General]
What is the purpose behind recommending hash routing with Wails? Are there any drawbacks to using the full URL instead of a hash strategy?  I've used both and tested various scenarios and haven't come across a reason why hash would be recommended over the full url.

**Comment by kafaa-dev** [comment]
At first, I was curious too, and because I've no reason to use hash routing, I used `window.location.pathname` for routing (browser routing/full URL).

But today, I've encountered this error:
```js
undefined is not an object (evaluating 'window['go']['main']')
```
The error sometimes occurs after a page reload, for example, when manually reloading or during hot-reloads.

Related issues:
- #2047 (error occured when accessing HTML files other than `index.html`)
- #2262, #2741 (error occured every time the app hot-reloads, and other cases)
- #2710 (error occured after `window.location.reload()`)

The cause is that the Wails Runtime and IPC are only injected when the path ends with `/index.html` or `/` (`/folder/index.html` probably works too) :
https://github.com/wailsapp/wails/blob/896a90f69ea46ec0695d0a975b82de9c1549e629/v2/pkg/assetserver/assetserver.go#L248-L255

The solution is either by using in-memory routing, hash routing, or manually inject the Wails IPC and Runtime (https://wails.io/docs/guides/frontend/#script-injection).

Hope this helps.
